plutus-core/plutus-ir/test/NamesSpec.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE TypeApplications #-}
+
+module NamesSpec
+    ( names
+    ) where
+
+import           PlutusIR.Generators.AST
+import           PlutusIR.Transform.Rename ()
+
+import           PlutusCore.Check.Scoping
+import           PlutusCore.Generators
+import           PlutusCore.Quote
+import           PlutusCore.Rename
+
+import           Control.Exception
+import           Hedgehog
+import           System.IO.Unsafe
+import           Test.Tasty
+import           Test.Tasty.Hedgehog
+
+scopingRename :: TestTree
+scopingRename =
+    testProperty "renaming does not destroy scoping" . withTests 1000 . property $ do
```

Comment:
Should we also include some of the bad renamer tests from PLC?

Reply:
Yep. No idea why I didn't do that.

Reply:
Ok, that one took me more time that I'm willing to admit, but it's done.